### PR TITLE
[FIX] Update EOS from config

### PR DESCRIPTION
**Pull request by zhengy001**
## Motivation
Correct EOS as default tokenizer `eos_token_id` may different from model config `eos_token_id`

## Checklist

- [x] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [ ] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [ ] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

**Comment by zhengy001** [comment]
Per @remixer-dec, model Mistral-Nemo-Instruct-2407-Q5_K_L.gguf has this issue.

File: python/sglang/srt/managers/tokenizer_manager.py [hunk 1]
```diff
@@ -247,6 +247,10 @@ async def _tokenize_one_request(
         # Parse sampling parameters
         sampling_params = SamplingParams(**obj.sampling_params)
         sampling_params.normalize(self.tokenizer)
+        sampling_params.update_from_config(
```
Comment (merrymercy) [inline python/sglang/srt/managers/tokenizer_manager.py]: Revert the changes in `tokenizer_manager.py` and `sampling_params.py`. Move all related changes to `schedule_batch.py`.
Reason:
The `sampling_params` is used to handle per-request configs, but this eos_token_ids form hf_config is the same for all requests, so we do not need to attach it to `sampling_params`.
To reduce the overhead, we should pre-process the hf-config and add some more conditions in `check_finished`

**Review by merrymercy — CHANGES_REQUESTED** [review]

File: python/sglang/srt/managers/tokenizer_manager.py (hunk 1 above)
Comment (zhengy001) [reply python/sglang/srt/managers/tokenizer_manager.py]: Add an input argument for `check_finished` as Req cannot acquire model info.

File: python/sglang/srt/managers/scheduler.py [hunk 2]
```diff
@@ -983,7 +983,7 @@ def process_batch_result_prefill(self, batch: ScheduleBatch, result):
 
                 if req.is_being_chunked <= 0:
                     req.output_ids.append(next_token_id)
-                    req.check_finished()
+                    req.check_finished(batch.get_hf_eos_token_id())
```
Comment (merrymercy) [inline python/sglang/srt/managers/scheduler.py]: revert all changes in this file

File: python/sglang/srt/managers/schedule_batch.py [hunk 3]
```diff
@@ -1133,6 +1138,9 @@ def get_model_worker_batch(self):
             input_embeds=self.input_embeds,
         )
 
+    def get_hf_eos_token_id(self) -> Optional[Set[int]]:
```
Comment (merrymercy) [inline python/sglang/srt/managers/schedule_batch.py]: This is still redundant. Can we cache this value `class Scheduler:` and pass it to `Req` here https://github.com/sgl-project/sglang/blob/dc3bee4815183d19761082b9ac90ada72d0151ef/python/sglang/srt/managers/scheduler.py#L497

**Review by merrymercy — CHANGES_REQUESTED** [review]
Left some final comments!

File: python/sglang/srt/managers/schedule_batch.py (hunk 3 above)
Comment (zhengy001) [reply python/sglang/srt/managers/schedule_batch.py]: Done

**Pull request merged**